notifier/src/test/kotlin/modules/JiraNotifierTest.kt

```diff
@@ -19,257 +19,195 @@
 
 package org.ossreviewtoolkit.notifier.modules
 
+import com.atlassian.jira.rest.client.api.IssueRestClient
+import com.atlassian.jira.rest.client.api.JiraRestClient
+import com.atlassian.jira.rest.client.api.OptionalIterable
+import com.atlassian.jira.rest.client.api.ProjectRestClient
+import com.atlassian.jira.rest.client.api.SearchRestClient
+import com.atlassian.jira.rest.client.api.domain.BasicIssue
+import com.atlassian.jira.rest.client.api.domain.Comment
 import com.atlassian.jira.rest.client.api.domain.Issue
+import com.atlassian.jira.rest.client.api.domain.IssueType
+import com.atlassian.jira.rest.client.api.domain.Project
+import com.atlassian.jira.rest.client.api.domain.SearchResult
+import com.atlassian.jira.rest.client.api.domain.Transition
 
-import com.github.tomakehurst.wiremock.WireMockServer
-import com.github.tomakehurst.wiremock.client.WireMock.aResponse
-import com.github.tomakehurst.wiremock.client.WireMock.equalTo
-import com.github.tomakehurst.wiremock.client.WireMock.exactly
-import com.github.tomakehurst.wiremock.client.WireMock.get
-import com.github.tomakehurst.wiremock.client.WireMock.matching
-import com.github.tomakehurst.wiremock.client.WireMock.post
-import com.github.tomakehurst.wiremock.client.WireMock.postRequestedFor
-import com.github.tomakehurst.wiremock.client.WireMock.urlEqualTo
-import com.github.tomakehurst.wiremock.client.WireMock.urlPathEqualTo
-import com.github.tomakehurst.wiremock.core.WireMockConfiguration
+import io.atlassian.util.concurrent.Promise
 
 import io.kotest.core.spec.style.WordSpec
 import io.kotest.matchers.result.shouldBeFailure
 import io.kotest.matchers.result.shouldBeSuccess
 import io.kotest.matchers.shouldBe
 import io.kotest.matchers.string.shouldContain
-import io.kotest.matchers.types.shouldBeTypeOf
 
-import java.io.File
-import java.net.URI
+import io.mockk.every
+import io.mockk.mockk
+import io.mockk.verify
 
-import org.ossreviewtoolkit.model.config.JiraConfiguration
+import java.net.URI
 
 class JiraNotifierTest : WordSpec({
-    val server = WireMockServer(
-        WireMockConfiguration.options()
-            .dynamicPort()
-            .usingFilesUnderDirectory(TEST_FILES_ROOT)
-    )
-
-    beforeSpec {
-        server.start()
-    }
-
-    afterSpec {
-        server.stop()
-    }
-
-    beforeEach {
-        server.resetAll()
-    }
-
     "createIssue" should {
         "succeed for valid input" {
-            val projectKey = "TEST"
-            val notifier = JiraNotifier(
-                JiraConfiguration("http://localhost:${server.port()}", "testuser", "testpassword")
-            )
+            val project = createProject()
```

Comment:
Looks good to me. Testing the notifier's interaction with the Jira client should be sufficient.

Reply:
Yes, that's the point, not testing the library but interaction with the library. And making it easier to add tests because I will add more functions to the `JiraNotifier`.